### DISCUSSION TITLE
Disclosure type errors

### DISCUSSION THREAD

**Discussion by liningpan** [tailwindlabs/headlessui#2775] [Help]
I observed some weird type errors related to disclosure when using pnpm with `hoist=false` and bazel. Adding the `preserveSymlinks` compiler option seem to resolve the issue. 

```json
{
  "compilerOptions": {
    "preserveSymlinks": true
  }
}
```

**Comment by jvliwanag** [comment]
@liningpan this has been driving me crazy the past few days. I still don't understand what's going on. but appreciate the workaround.

it's not limited to disclosure -- all components seem to not resolve the correct react types.

**Comment by jvliwanag** [comment]
Looks like this is due to having different @types/react. Can be worked around by manually overriding. For instance in pnpm:

```
"pnpm": {
    "packageExtensions": {
      "@headlessui/react": {
        "peerDependencies": {
          "@types/react": "*"
        }
      }
    }
  }
```
